csrc/core/CachingAllocatorHelper.h

```diff
@@ -21,9 +21,8 @@ class CachingAllocatorHelper {
   // Synchronizes the stream. e.g. cudaStreamSynchronize
   virtual int synchronizeStream(void* stream) = 0;
 
-  // Wait for compute device to finish. e.g. cudaDeviceSynchronize. When
-  // checkError is true, it will check error, and throw error if there is.
-  virtual void deviceSynchronize(bool checkError) = 0;
+  // Wait for compute device to finish. e.g. cudaDeviceSynchronize.
+  virtual void deviceSynchronize() = 0;
```

Comment:
```suggestion
  virtual void synchronizeDevice() = 0;
```
可以和上面synchronizeStream以及它转调的synchronize_device命名顺序保持一致，顺序不一样容易误解成是两个方法

Reply:
佳伟建议c10_npu内的接口还用device_synchronize，所以这里就保持相同了